### PR TITLE
Alter member search to be more inclusive

### PR DESCRIPTION
**Pull request by tnorthcutt**
This change mimics WP core's method of prepending and appending `*` to user searches, so that when the `WP_User_Query` is prepared, the `LIKE` clauses will be surrounded by `%`. This results in finding e.g. a member with the email address `john123@gmail.com` when searching for `john`; previously this member would not be found.

Closes https://github.com/restrictcontentpro/restrict-content-pro/issues/1679

### CODE REVIEW

File: includes/admin/members/members-page.php [hunk 1]
```diff
@@ -29,6 +29,10 @@ function rcp_members_page() {
 			$order           = ! empty( $_GET['order']  )  ? urldecode( $_GET['order']  ) : 'DESC';
 			$search          = ! empty( $_GET['s'] )       ? urldecode( $_GET['s'] )      : '';
 
+			if ( '' !== $search ) {
+			    $search = '*' . $search . '*';
+            }
```
Comment (pippinsplugins) [inline includes/admin/members/members-page.php]: Looks like a minor indention hiccup here :)

Comment (tnorthcutt) [reply includes/admin/members/members-page.php]: Good catch! Good old tabs/spaces got me.

File: includes/admin/members/members-page.php [hunk 2]
```diff
@@ -102,7 +106,7 @@ function rcp_members_page() {
 			</ul>
 			<form id="rcp-member-search" method="get" action="<?php menu_page_url( 'rcp-members' ); ?>">
 				<label class="screen-reader-text" for="rcp-member-search-input"><?php _e( 'Search Members', 'rcp' ); ?></label>
-				<input type="search" id="rcp-member-search-input" name="s" value="<?php echo esc_attr( $search ); ?>"/>
+                <input type="search" id="rcp-member-search-input" name="s" value="<?php echo isset($_REQUEST['s']) ? esc_attr( wp_unslash( $_REQUEST['s'] ) ) : ''; ?>"/>
```
Comment (nosegraze) [inline includes/admin/members/members-page.php]: This is indented with spaces but it should be tabs.

Comment (tnorthcutt) [reply includes/admin/members/members-page.php]: Sorry about that. Fixed up now.

**Comment by tnorthcutt** [comment]
Thanks @mindctrl. I'm not familiar with searching by payment profile. Can you give me a little more info/an example to reproduce with? I'd be happy to fix it up.

**Comment by mindctrl** [comment]
@tnorthcutt you can search members by their payment profile ID, which you can see on the member edit screen. In the case of Stripe, it's their customer number.

image

You can search for `payment_profile:xxxxxxxxx` to find the member with that profile ID.

**Comment by tnorthcutt** [comment]
@mindctrl Ah, thanks! This was also breaking searches for `first_name:` and `last_name:`. I moved to altering the query to add asterisks after we check for those special cases.

**Pull request merged**